### PR TITLE
ASF Nexus SNAPSHOT publish

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

https://infra.apache.org/publishing-maven-artifacts.html
https://issues.apache.org/jira/browse/INFRA-22114

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
awesome！

**Comment by pan3793** [comment]
Successfully deployed the SNAPSHOT version to https://repository.apache.org/content/repositories/snapshots/org/apache/kyuubi/,
and we can see kyuubi group in https://repository.apache.org/ (have not test staging deploy)

**Comment by pan3793** [comment]
Deploy SNAPSHOT process: https://infra.apache.org/publishing-maven-artifacts.html

TL;DR
1. setup gpg and maven auth
```
<settings>
...
  <servers>
    <!-- To publish a snapshot of your project -->
    <server>
      <id>apache.snapshots.https</id>
      <username> <!-- YOUR APACHE LDAP USERNAME --> </username>
      <password> <!-- YOUR APACHE LDAP PASSWORD (encrypted) --> </password>
    </server>
    <!-- To stage a release of your project -->
    <server>
      <id>apache.releases.https</id>
      <username> <!-- YOUR APACHE LDAP USERNAME --> </username>
      <password> <!-- YOUR APACHE LDAP PASSWORD (encrypted) --> </password>
    </server>
   ...
  </servers>
</settings>
```
2. `mvn deploy -Papache-release,kyuubi-extension-spark_3.1,spark-block-cleaner,tpcds -DskipTests`

I'm going to learn how other mature TLP (i.e. spark, flink) do the release, and a serial of following PRs will be sent in next few days.

**Review by ulysses-you — APPROVED** [review]
great work!

**Review by yaooqinn — APPROVED** [review]
LGTM！

**Comment by yaooqinn** [comment]
thanks, merged to master
